awx/main/models/events.py

```diff
@@ -546,6 +546,17 @@ def _update_host_summary_from_stats(self, hostnames):
 
             Host.objects.bulk_update(list(updated_hosts), ['last_job_id', 'last_job_host_summary_id'], batch_size=100)
 
+            updated_hosts_list = list()
+            for host in updated_hosts:
+                updated_hosts_list.append(host.name)
+
+            # bulk-create
+            current_time = now()
+            HostMetrics.objects.bulk_create(
+                [HostMetrics(hostname=hostname, last_automation=current_time) for hostname in updated_hosts_list], ignore_conflicts=True
```

Comment:
We want to pull the `updated_hosts_list` from `hostnames` that is passed into the function https://github.com/ansible/awx/blob/devel/awx/main/models/events.py#L518 

This list potentially contains **more** hosts than the host summary fields list because it can contain host names that aren't in the AWX database.  Below is an add_host example you can use to test on your current version vs. a version that uses `hostnames`

```
- hosts: all
  gather_facts: false
  tasks:
    - name: "Do something"
      shell: ls

    - add_host:
        name: "boo1"
        ansible_connection: "local"

- hosts: all
  gather_facts: false
  tasks:

    - name: "Do something else"
      shell: ls
```

```
# Make an awx inventory that uses two hosts
[all]
fake1 ansible_connection=local
fake2 ansible_connection=local
```


```
ansible-playbook -i inventory main.yml

PLAY [all] ************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************

TASK [Do something] ***************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************
[DEPRECATION WARNING]: Distribution ubuntu 20.04 on host fake1 should use /usr/bin/python3, but is using /usr/bin/python for backward compatibility with prior Ansible releases. A future Ansible release will default to using the discovered platform python for this host. See https://docs.ansible.com/ansible/2.9/reference_appendices/interpreter_discovery.html for more information. This feature will be removed in
version 2.12. Deprecation warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.
changed: [fake1]
[DEPRECATION WARNING]: Distribution ubuntu 20.04 on host fake2 should use /usr/bin/python3, but is using /usr/bin/python for backward compatibility with prior Ansible releases. A future Ansible release will default to using the discovered platform python for this host. See https://docs.ansible.com/ansible/2.9/reference_appendices/interpreter_discovery.html for more information. This feature will be removed in
version 2.12. Deprecation warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.
changed: [fake2]

TASK [add_host] *******************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************
changed: [fake1]

PLAY [all] ************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************

TASK [Do something else] **********************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************
changed: [fake2]
changed: [fake1]
[DEPRECATION WARNING]: Distribution ubuntu 20.04 on host boo1 should use /usr/bin/python3, but is using /usr/bin/python for backward compatibility with prior Ansible releases. A future Ansible release will default to using the discovered platform python for this host. See https://docs.ansible.com/ansible/2.9/reference_appendices/interpreter_discovery.html for more information. This feature will be removed in
version 2.12. Deprecation warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.
changed: [boo1]

PLAY RECAP ************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************
boo1                       : ok=1    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
fake1                      : ok=3    changed=3    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
fake2                      : ok=2    changed=2    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
```
In the above output, `boo1` is the host dynamically added to the ansible runtime inventory. The "play recap" is basically equivalent to the `playbook_on_stats`.

Reply:
I have made the required changes.